cmd/create/accountroles/cmd.go

```diff
@@ -358,120 +360,144 @@ func run(cmd *cobra.Command, argv []string) {
 	}
 }
 
-func buildCommands(prefix string, permissionsBoundary string, accountID string, defaultPolicyVersion string,
-	path string, managedPolicies bool, policies map[string]*cmv1.AWSSTSPolicy) (string, error) {
-	commands := []string{}
+type creator interface {
+	createRoles(*rosa.Runtime, rolesCreationInput) error
+	getRoleTags(string, rolesCreationInput) map[string]string
+	buildCommands(rolesCreationInput) (string, error)
+}
+
+func initCreator(managedPolicies bool) creator {
+	if managedPolicies {
+		return managedPoliciesCreator{}
+	}
+
+	return unmanagedPoliciesCreator{}
+}
+
+func buildRolesCreationInput(prefix, permissionsBoundary, accountID, env string,
+	policies map[string]*cmv1.AWSSTSPolicy, defaultPolicyVersion string,
+	path string) rolesCreationInput {
+	return rolesCreationInput{
+		prefix:               prefix,
+		permissionsBoundary:  permissionsBoundary,
+		accountID:            accountID,
+		env:                  env,
+		policies:             policies,
+		defaultPolicyVersion: defaultPolicyVersion,
+		path:                 path,
+	}
+}
+
+type rolesCreationInput struct {
+	prefix               string
+	permissionsBoundary  string
+	accountID            string
+	env                  string
+	policies             map[string]*cmv1.AWSSTSPolicy
+	defaultPolicyVersion string
+	path                 string
+}
+
+type managedPoliciesCreator struct{}
+
+func (mp managedPoliciesCreator) createRoles(r *rosa.Runtime, input rolesCreationInput) error {
```

Comment:
`mp *managedPoliciesCreator`

Reply:
`input *rolesCreationInput` no need to copy by value the entire struct

Reply:
Fixed.